### PR TITLE
feat: Add tiddit

### PR DESCRIPTION
**Pull request by zezzipa**
Adding tiddit (https://github.com/SciLifeLab/TIDDIT), calling sv, getting vcf output.

### This PR:

(If this is a release PR, no need to add following. Leave this part empty)
(Use the following lines to create a PR text body. Make sure to remove all non-relevant one after you're done)
(Repeat each field as many times as necessary)

Added: for new features.
Changed: for changes in existing functionality.
Deprecated: for soon-to-be removed features.
Removed: for now removed features.
Fixed: for any bug fixes.
Security: in case of vulnerabilities.

### Review and tests: 
- [ ] Tests pass
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Code review
- [ ] `CHANGELOG.md` is updated
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: workflow/rules/tiddit.smk [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+rule tiddit:
+    input:
+        bam="alignment/samtools_merge_bam/{sample}_{type}.bam",
+    output:
+        vcf="cnv_sv/tiddit/{sample}_{type}.vcf"
+    params:
+        extra=config.get("tiddit", {}).get("extra", ""),
+        outdir="cnv_sv/tiddit/{sample}_{type}",
+    log:
+        "cnv_sv/tiddit/{sample}_{type}.log"
```
Comment (jonca79) [inline workflow/rules/tiddit.smk:16]:
```suggestion
        "cnv_sv/tiddit/{sample}_{type}.vcf.log"
```

File: workflow/rules/tiddit.smk [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        threads=config.get("tiddit", {}).get("threads", config["default_resources"]["threads"]),
+        time=config.get("tiddit", {}).get("time", config["default_resources"]["time"]),
+    container:
+        config.get("tiddit", {}).get("container", config["default_container"])
+    conda:
+        "../envs/tiddit.yaml"
+    message:
+       "{rule}: Run tiddit on {wildcards.sample}_{wildcards.type}"
+    shell:
+        "TIDDIT.py --sv --bam {input.bam} -o {params.outdir} &> {log}"
```
Comment (jonca79) [inline workflow/rules/tiddit.smk:34]:
```suggestion
        "TIDDIT.py "
        "--sv "
        "--bam {input.bam} "
        "-o {params.outprefix} &> {log}"
```

File: workflow/rules/tiddit.smk [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+
+rule tiddit:
+    input:
+        bam="alignment/samtools_merge_bam/{sample}_{type}.bam",
+    output:
+        vcf="cnv_sv/tiddit/{sample}_{type}.vcf"
+    params:
+        extra=config.get("tiddit", {}).get("extra", ""),
+        outdir="cnv_sv/tiddit/{sample}_{type}",
```
Comment (jonca79) [inline workflow/rules/tiddit.smk:14]:
```suggestion
        outprefix="cnv_sv/tiddit/{sample}_{type}",
```

File: workflow/rules/common.smk [hunk 4]
```diff
@@ -60,6 +60,7 @@ def compile_output_list(wildcards):
         "cnv_sv/svdb_query": ["svdb_query.vcf"],
         "cnv_sv/exomedepth_call": ["SV.txt"],
         "cnv_sv/pindel_vcf": ["vcf"],
+        "cnv_sv/tiddit": ["vcf",],
```
Comment (jonca79) [inline workflow/rules/common.smk:63]:
```suggestion
        "cnv_sv/tiddit": ["vcf"],
```

File: workflow/rules/tiddit.smk [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+__copyright__ = "Copyright 2022, Jessika Nordin"
+__email__ = "jessika.nordin@scilifelab.uu.se"
+__license__ = "GPL-3"
+
+
+rule tiddit:
+    input:
+        bam="alignment/samtools_merge_bam/{sample}_{type}.bam",
+    output:
+        vcf="cnv_sv/tiddit/{sample}_{type}.vcf"
```
Comment (jonca79) [inline workflow/rules/tiddit.smk:11]:
```suggestion
        vcf="cnv_sv/tiddit/{sample}_{type}.vcf",
```

File: workflow/rules/tiddit.smk [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    params:
+        extra=config.get("tiddit", {}).get("extra", ""),
+        outprefix="cnv_sv/tiddit/{sample}_{type}",
+    log:
+        "cnv_sv/tiddit/{sample}_{type}.vcf.log"
```
Comment (jonca79) [inline workflow/rules/tiddit.smk:16]:
```suggestion
        "cnv_sv/tiddit/{sample}_{type}.vcf.log",
```

File: workflow/rules/tiddit.smk [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        mem_per_cpu=config.get("tiddit", {}).get("mem_per_cpu", config["default_resources"]["mem_per_cpu"]),
+        partition=config.get("tiddit", {}).get("partition", config["default_resources"]["partition"]),
+        threads=config.get("tiddit", {}).get("threads", config["default_resources"]["threads"]),
+        time=config.get("tiddit", {}).get("time", config["default_resources"]["time"]),
+    container:
+        config.get("tiddit", {}).get("container", config["default_container"])
+    conda:
+        "../envs/tiddit.yaml"
+    message:
+       "{rule}: Run tiddit on {wildcards.sample}_{wildcards.type}"
```
Comment (jonca79) [inline workflow/rules/tiddit.smk:31]:
```suggestion
        "{rule}: Run tiddit on {wildcards.sample}_{wildcards.type}"
```

**Comment by jonca79** [comment]
Test set is missing under .tests/integration/. If small test files are hard to make you can use empty dummy files.

**Comment by jonca79** [comment]
Also missing schemas (config and resources) for the rule.

**Comment by zezzipa** [comment]
> Also missing schemas (config and resources) for the rule.
Added that

> Test set is missing under .tests/integration/. If small test files are hard to make you can use empty dummy files.
How small does it have to be? An empty file will cause the rule to fail, and I guess that is not good?

File: workflow/rules/tiddit.smk [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+__copyright__ = "Copyright 2022, Jessika Nordin"
+__email__ = "jessika.nordin@scilifelab.uu.se"
+__license__ = "GPL-3"
+
+
+rule tiddit:
+    input:
+        bam="alignment/samtools_merge_bam/{sample}_{type}.bam",
+    output:
+        vcf="cnv_sv/tiddit/{sample}_{type}.vcf",
```
Comment (jonca79) [inline workflow/rules/tiddit.smk:11]:
```suggestion
        bam=temp("cnv_sv/tiddit/{sample}_{type}.bam"),
        ploidy=temp("cnv_sv/tiddit/{sample}_{type}.ploidy.tab"),
        signals=temp("cnv_sv/tiddit/{sample}_{type}.signals.tab"),
        vcf=temp("cnv_sv/tiddit/{sample}_{type}.vcf"),
        wig=temp("cnv_sv/tiddit/{sample}_{type}.wig"),
```

**Review by jonca79 — APPROVED** [review]

**Pull request merged**